tables_app/src/androidTest/java/org/opendatakit/espresso/TableDisplayActivityTest.java

```diff
@@ -0,0 +1,99 @@
+package org.opendatakit.espresso;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import androidx.fragment.app.FragmentManager;
+import androidx.fragment.app.FragmentTransaction;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.espresso.contrib.RecyclerViewActions;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;
+import androidx.test.rule.GrantPermissionRule;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.junit.Assert.assertTrue;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.opendatakit.tables.R;
+import org.opendatakit.tables.activities.TableDisplayActivity;
+import org.opendatakit.tables.fragments.TableManagerFragment;
+
+@RunWith(AndroidJUnit4ClassRunner.class)
+public class TableDisplayActivityTest {
+
+    @Rule
+    public GrantPermissionRule permissionRule = GrantPermissionRule.grant(
+            android.Manifest.permission.READ_EXTERNAL_STORAGE,
+            android.Manifest.permission.WRITE_EXTERNAL_STORAGE);
+
+    @Rule
+    public ActivityScenarioRule<TableDisplayActivity> activityRule =
+            new ActivityScenarioRule<>(TableDisplayActivity.class);
+
+    private ActivityScenario<TableDisplayActivity> scenario;
+
+    @Before
+    public void setUp() {
+        scenario = activityRule.getScenario();
+    }
+
+    @After
+    public void tearDown() {
+        // Clean up after tests
+        scenario.close();
+    }
+
+    @Test
+    public void testOnCreate() {
+        // Verify the UI elements and initial state
+        onView(withId(R.id.recycler_view)).check(matches(isDisplayed()));
+    }
+
+
+    @Test
+    public void testFragmentTransaction() {
```

Comment:
@Chinex-Boroja Manually adding the framgment here, does not fully capture the intent of testing. When you manually add the fragment, it’s almost guaranteed that it will be added, which may not provide much value in terms of testing.

The test would be more meaningful if it verified that the fragment was added as part of the actual UI flow, triggered by user interaction or lifecycle events, rather than by directly invoking the fragment transaction in the test. This way, the test would better reflect real-world scenarios and provide more meaningful insights.

If you need more clarifications, I am happy to huddle.
CC: @wbrunette what are your thoughts?